File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/base/exceptions/ConnectionException.java
package com.hgianastasio.biblioulbrav2.core.base.exceptions;

/**
 * Created by heitorgianastasio on 4/26/17.
 */

public class ConnectionException extends RuntimeException {
    public ConnectionException() {
    }

    public ConnectionException(String s) {
        super(s);
    }

    public ConnectionException(String s, Throwable throwable) {
        super(s, throwable);
    }

    public ConnectionException(Throwable throwable) {
        super(throwable);
    }

    @Override
    public String getMessage() {
        return "Não foi possível acessar o servidor";
    }
}

File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/base/exceptions/ConnectionException.kt
package com.hgianastasio.biblioulbrav2.core.base.exceptions

/**
 * Created by heitorgianastasio on 4/26/17.
 */
class ConnectionException : RuntimeException {
    constructor() {}
    constructor(s: String?) : super(s) {}
    constructor(s: String?, throwable: Throwable?) : super(s, throwable) {}
    constructor(throwable: Throwable?) : super(throwable) {}

    override val message: String
        get() = "Não foi possível acessar o servidor"
}